### PR TITLE
Add rails 5.1 support

### PR DESCRIPTION
**Pull request by will89**

### CODE REVIEW

**Review by jturkel — APPROVED** [review]
Glad to see the Travis tests were passing with Rails 5.1 prior to this change ;)

**Comment by will89** [comment]
@jturkel Want to take another look? I updated this to use appraisals for the build matrix. Turns out I had the gemspec setup wrong, I failed to use the env vars in it. :)

File: activerecord-forbid_implicit_connection_checkout.gemspec [hunk 1]
```diff
@@ -29,11 +28,12 @@ Gem::Specification.new do |spec|
   spec.bindir        = 'bin'
   spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
   spec.require_paths = ['lib']
-  spec.add_dependency 'activerecord', ['>= 5', '< 5.1']
-  spec.add_dependency 'activemodel', ['>= 5', '< 5.1']
-  spec.add_dependency 'activesupport', ['>= 5', '< 5.1']
+  spec.add_dependency 'activerecord', ['>= 5', '< 5.2']
```
Comment (jturkel) [inline activerecord-forbid_implicit_connection_checkout.gemspec]: The version constraints shouldn't be wrapped in an array i.e. it should be 

```
spec.add_dependency 'activerecord', '>= 5', '< 5.2'
```

File: Appraisals [hunk 2]
```diff
@@ -0,0 +1,9 @@
+appraise 'rails-5.0' do
```
Comment (jturkel) [inline Appraisals:1]: Should we include `activemodel` in these `appraise` blocks too?

**Review by jturkel — APPROVED** [review]
A few small comments but looks good

**Pull request merged**